### PR TITLE
[doc] add example for SubgraphSampler

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/subgraph_sampler.py [hunk 1]
```diff
@@ -148,7 +151,25 @@ def _sample_subgraphs(self, seeds):
...
+        >>>         self.fanouts = fanouts
+        >>>     def sample_subgraphs(self, seeds):
+        >>>         # Sample subgraphs from the given seeds.
+        >>>         subgs = []
+        >>>         for fanout in reversed(self.fanouts):
+        >>>             # Sample a fixed number of neighbors for each seed node.
+        >>>             input_nodes, subg = self.graph.sample_neighbors(seeds, fanout)
+        >>>             seeds = input_nodes
+        >>>             subgs.insert(0, subg)
+        >>>         return input_nodes, subgs
```
Comment (frozenbugs) [inline python/dgl/graphbolt/subgraph_sampler.py:173]:
```suggestion
        >>>         # Sample subgraphs from the given seeds.
        >>>         subgraphs = []
        >>>         subgraphs_nodes = {}
        >>>         for fanout in reversed(self.fanouts):
        >>>             subgraph = self.graph.sample_neighbors(seeds, fanout)
        >>>             subgraphs.insert(0, subgraph)
        >>>             subgraphs_nodes.update(subgraph.nodes)
        >>>             seeds = subgraph.nodes
        >>>         return subgraphs.nodes, subgraphs
```

Comment (Rhett-Ying) [reply python/dgl/graphbolt/subgraph_sampler.py:173]: As `input_nodes` is confusing, then we should only return subgraphs only for generality.

Comment (frozenbugs) [reply python/dgl/graphbolt/subgraph_sampler.py:173]: But we need subgraphs_nodes for feature fetching in later stage.

Comment (frozenbugs) [reply python/dgl/graphbolt/subgraph_sampler.py:173]: The API here could be further polished.

Comment (Rhett-Ying) [reply python/dgl/graphbolt/subgraph_sampler.py]: all right.

**Review by frozenbugs — APPROVED** [review]

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**